broadinstitute/scp-ingest-pipeline
# Remove cell_type requirement for cell_type__custom (SCP-4835)

This PR removes the declared dependency of `cell_type__custom` on the presence of a corresponding `cell_type` entry in SCP metadata files. When we originally envisioned `cell_type__custom`, we assumed study owners would always be able to find an acceptable “less granular” ontology term to describe cells that needed a custom label. From [Zendesk #298615](https://broadinstitute.zendesk.com/agent/tickets/298615) we learned that study owners would rather not assign a more general `cell_type` annotation for their custom cells and would rather omit a `cell_type` designation for those cells.

An additional automated check in `test_validate_metadata` under `test_custom` automatically tests that this more flexible metadata schema change is adhered to.

To test manually:

Prior state - in the development branch (or prod), see that upload of the file [scp-ingest-pipeline/tests/data/annotation/metadata/convention/valid_cell_type__custom_v2.2.1.txt](https://github.com/broadinstitute/scp-ingest-pipeline/blob/jlc_ctcustom_nodep/tests/data/annotation/metadata/convention/valid_cell_type__custom_v2.2.1.txt) fails validation.

Confirm change - check out this branch and either:

1. Run in your local portal instance - set your local instance to use the following docker image:
`gcr.io/broad-singlecellportal-staging/scp-ingest-jlc_ctcustom_nodep:a177238`
(in Admin config, add `Ingest Pipeline Docker Image`, value type `string` with the above value)
OR
2. Run ingest pipeline via command line
a. Activate the scp-ingest-pipeline repo virtualenv
b. From the ingest directory of the scp-ingest-pipeline repo, perform this setup:
`source ../scripts/setup_mongo_dev.sh <path to your Github token file>`

If you get an authentication error from gcloud, run the following:
`gcloud auth application-default login --no-launch-browser `

c. Run this metadata ingest job from the ingest directory of the scp-ingest-pipeline repo:

```
python ingest_pipeline.py --study-id 5d276a50421aa9117c982845 --study-file-id 5dd5ae25421aa910a723a337 ingest_cell_metadata --cell-metadata-file ../tests/data/annotation/metadata/convention/valid_cell_type__custom_v2.2.1.txt --study-accession SCP4835 --ingest-cell-metadata --validate-convention --bq-dataset cell_metadata_test --bq-table alexandria_convention
```

This fulfills SCP-4835.
